source/includes/write/transaction.scala

```diff
@@ -0,0 +1,68 @@
+import org.mongodb.scala._
+import java.util.concurrent.TimeUnit
+import scala.concurrent.Await
+import org.mongodb.scala.model.Updates._
+import org.mongodb.scala.model.Filters._
+import scala.concurrent.duration.Duration
+import org.mongodb.scala.result._
+
+object Transaction {
+// begin-transaction-method
+  def runTransaction(
+      database: MongoDatabase,
+      observable: SingleObservable[ClientSession]
+  ): SingleObservable[ClientSession] = {
+    observable.map(clientSession => {
+      val moviesCollection = database.getCollection("movies")
+      val usersCollection = database.getCollection("users")
+
+      val transactionOptions = TransactionOptions
+        .builder()
+        .readPreference(ReadPreference.primary())
+        .readConcern(ReadConcern.SNAPSHOT)
+        .writeConcern(WriteConcern.MAJORITY)
+        .build()
+
+      // Starts the transaction with specified options
+      clientSession.startTransaction(transactionOptions)
+
+      // Inserts a document into the "movies" collection  
+      moviesCollection
+        .insertOne(
+          clientSession,
+          Document("name" -> "The Menu", "runtime" -> 107)
+        )
+        .subscribe((res: InsertOneResult) => println(res))
+
+      // Updates a document in the "users" collection
+      usersCollection
+        .updateOne(clientSession, equal("name", "Amy Phillips"), set("name", "Amy Ryan"))
+        .subscribe((res: UpdateResult) => println(res))
+
+      clientSession
```

Comment:
Confirming this line is intentional? It looks like it could be out of place (but idk Scala)

Reply:
Yes! That's the return value